src/ks_thread.c

```diff
@@ -389,7 +413,12 @@ static ks_status_t __init_os_thread(ks_thread_t *thread)
 	}
 #endif
 
+	ks_mutex_lock(thread->mutex);
```

Comment:
What are you protecting against here with that lock?

Reply:
Discussed verbaly